pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -107,6 +108,64 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// Finalizer name
+	userAccFinalizerName := "finalizer.toolchain.dev.openshift.com"
+
+	// Add the finalizer if it is not present
+	if !containsString(userAcc.ObjectMeta.Finalizers, userAccFinalizerName) {
+		userAcc.ObjectMeta.Finalizers = append(userAcc.ObjectMeta.Finalizers, userAccFinalizerName)
+		if err := r.client.Update(context.TODO(), userAcc); err != nil {
+			return reconcile.Result{Requeue: true}, err
```

Comment:
Minor: No need to set `Requeue: true`.
This should work `return reconcile.Result{}, err`.  Why?? bcoz, we did `Update` to userAcc so `Reconcile()` will be called back for the same.

Reply:
Thanks Nurali - addressed as part of d8498c421eaf5dd07ecf330e6cdd4063ce57e21d